app/src/main/java/com/asfoundation/wallet/ui/transact/TransferFragment.kt

```diff
@@ -92,12 +96,12 @@ class TransferFragment : DaggerFragment(), TransferFragmentView {
                                         amount: BigDecimal): Completable {
 
     return defaultTokenInfoProvider.defaultToken.doOnSuccess {
-      val transaction = TransactionBuilder(it)
-      transaction.amount(amount)
-      transaction.toAddress(toWalletAddress)
-      transaction.fromAddress(walletAddress)
-      confirmationRouter.open(activity, transaction)
-    }
+          val transaction = TransactionBuilder(it)
```

Comment:
`  override fun openAppcConfirmationView(walletAddress: String, toWalletAddress: String,
                                        amount: BigDecimal): Completable {

    return defaultTokenInfoProvider.defaultToken.doOnSuccess {
      with(TransactionBuilder(it)) {
        amount(amount)
        toAddress(toWalletAddress)
        fromAddress(walletAddress)
        confirmationRouter.open(activity, this)
      }
    }
        .ignoreElement()
  }`

This is an alternative way. do you like it?